# DSAI LAB 5 Proshita Agarwal

## Part 1:

In [1]:
import pandas as pd
import numpy as np

df = pd.read_csv("customer_dataset.csv")
df.head()

,Customer ID,Purchase Method,Membership Card,Total Spend,Average Discount
0,1,Online,Gold,142,12
1,2,In-store,Silver,89,8
2,3,In-store,Silver,47,3
3,4,Online,Gold,230,18
4,5,Online,Bronze,32,5


In [3]:
# Overall mean of Total Spend
overall_mean = df["Total Spend"].mean()
print("Overall Mean (Total Spend):", overall_mean)

Overall Mean (Total Spend): 96.79


In [5]:
# Group by Membership Card type
group_stats = df.groupby("Membership Card")["Total Spend"].agg(["mean", "var", "count"])
print(group_stats)

                       mean          var  count
Membership Card                                
Bronze            43.153846   252.455385     26
Gold             147.850000  6223.361538     40
Silver            77.735294  1289.170232     34


In [6]:
# Calculating SSG
SSG = sum(group_stats["count"] * (group_stats["mean"] - overall_mean)**2)
print("SSG (Between Groups):", SSG)

SSG (Between Groups): 191427.48773755657


In [9]:
# Calculating SSE
SSE = 0
for name, group in df.groupby("Membership Card"):
    group_mean = group["Total Spend"].mean()
    SSE += sum((group["Total Spend"] - group_mean)**2)

print("SSE (Within Groups):", SSE)

SSE (Within Groups): 291565.10226244346


In [10]:
# Number of groups (k) and total observations (N)
k = df["Membership Card"].nunique()
N = len(df)

# Degrees of freedom
df_between = k - 1
df_within = N - k

print("df_between:", df_between)
print("df_within:", df_within)

df_between: 2
df_within: 97


In [12]:
#Mean squares
MSG = SSG / df_between
MSE = SSE / df_within

print("MSG (Mean Square Between Groups):", MSG)
print("MSE (Mean Square Within Groups):", MSE)

MSG (Mean Square Between Groups): 95713.74386877829
MSE (Mean Square Within Groups): 3005.8257965200355


In [13]:
#f stat
F_statistic = MSG / MSE
print("F-statistic:", F_statistic)

F-statistic: 31.842744838902476


f-stat critical value at alpha = 0.05 and dof = (2,97), from f table = 3.087, now calulated value of f stat is much greater than the critical value(31.84>3.087). since calculated value> critical value, therefore, we reject H0. There are signifiacnt changes the amount spend depending upon membership level

### PART 2:

In [15]:
# grouping dataset by membership card type
groups = [group["Total Spend"].values for name, group in df.groupby("Membership Card")]
for i, g in enumerate(groups, 1):
    print(f"Group {i} sample size: {len(g)}")

Group 1 sample size: 26
Group 2 sample size: 40
Group 3 sample size: 34


In [16]:
print("Null Hypothesis (H0): There are no significant differences in spending among the membership types.")
print("Alternative Hypothesis (H1): There are significant differences in spending among the membership types.")

Null Hypothesis (H0): There are no significant differences in spending among the membership types.
Alternative Hypothesis (H1): There are significant differences in spending among the membership types.


In [17]:
from scipy.stats import f_oneway

# One-way ANOVA
anova_result = f_oneway(*groups)
print(anova_result)

F_onewayResult(statistic=np.float64(31.84274483890246), pvalue=np.float64(2.336474646572437e-11))


in the above code, groups is actually a list of arrays so i used * which automatically unpacks the list so each array is passed as its own argument. asterisk here is an unpacking operator

In [18]:
F_statistic = anova_result.statistic
p_value = anova_result.pvalue

print("F-statistic:", F_statistic)
print("p-value:", p_value)

F-statistic: 31.84274483890246
p-value: 2.336474646572437e-11


In [19]:
alpha = 0.05  # significance level
if p_value < alpha:
    print(f"Since p-value ({p_value:.4f}) < 0.05, we reject H0.")
    print("Conclusion: There are significant differences in spending among the membership types.")
else:
    print(f"Since p-value ({p_value:.4f}) >= 0.05, we fail to reject H0.")
    print("Conclusion: There are no significant differences in spending among the membership types.")

Since p-value (0.0000) < 0.05, we reject H0.
Conclusion: There are significant differences in spending among the membership types.


In this lab, we analyzed customer spending data across different membership card types using one-way ANOVA.

Part 1 (Manual Computation):
We calculated the overall mean, group-wise means and variances, and then computed SSG (between groups) and SSE (within groups). From these, we derived the MSG, MSE, and F-statistic. A relatively high F-statistic value suggests potential differences in average spending across membership types.

Part 2 (SciPy ANOVA):
Using scipy.stats.f_oneway, we obtained the F-statistic and p-value directly.

Null Hypothesis (H0): No significant difference in spending among membership card types.

Alternative Hypothesis (H1): At least one group differs significantly.

Interpretation:
Based on the p-value at α = 0.05, we decide whether to reject H0.

If p < 0.05 → reject H0 → significant differences exist.

If p ≥ 0.05 → fail to reject H0 → no significant differences found.

Thus, the lab demonstrates both the manual calculation process of ANOVA and its practical application using statistical libraries.